elasticsearch/client.go

```diff
@@ -153,16 +175,23 @@ func (cli *Client) callElastic(ctx context.Context, path, method string, payload
 
 	jsonBody, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		log.Event(ctx, "failed to read response body from call to elastic", log.ERROR, log.Error(err), logData)
-		return nil, resp.StatusCode, err
+		log.Error(ctx, "failed to read response body from call to elastic", err)
+		return jsonBody, resp.StatusCode, err
 	}
+
 	logData["json_body"] = string(jsonBody)
 	logData["status_code"] = resp.StatusCode
 
 	if resp.StatusCode < http.StatusOK || resp.StatusCode >= 300 {
-		log.Event(ctx, "failed", log.ERROR, log.Error(ErrorUnexpectedStatusCode), logData)
-		return nil, resp.StatusCode, ErrorUnexpectedStatusCode
+		log.Error(ctx, "failed as unexpected code", ErrorUnexpectedStatusCode)
```

Comment:
Is there a reason why we are checking a range instead of a specific status?

Reply:
this is the existing implementation, handling of 500 is part of this PR

Reply:
So I understand the range and returning status code.

1. Any reason for removing logData from error log?

2. Why have you implemented a check on 500 status code?

3. And then you have added a log.Info event? What value does this add?

4. Why are we returning the jsonBody on lines 191?

5. If we are keeping the check on status 500, why are we not using a constant for the error in keeping with the library code? e.g. `ErrInternalServer` and what about other 5** status codes?

Reply:
1. logData reintroduced
2. Two tests, in particular, were in mind, in addition to range of specific status, network error and once you reach the es-server so possible server error(500). look at my previous comments about jsonbody/status as above.
3. log.info removed
4. Introduced ErrInternalServer but for other 5** status, these codes can be captured once we reach the server is ready to serve.

Reply:
Sorry but im not sure the code has been updated to match your response. Please check